jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSMessageListener.java

```diff
@@ -0,0 +1,140 @@
+    
+    /**
+     * A method used to setup the MessageListener for the given destination (topic or queue)
+     * @param connectionFactoryName     The name of the connection factory used to connect to the JMS Server
+     * @param dest                      The name of the destination to connect to (topic or queue)
+     * @param isQueue                   A boolean flag used to create the appropriate type of destination (queue or topic)
+     * @param username                  The username used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @param password                  The password used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public void setupMessageListener(String connectionFactoryName, String dest, boolean isQueue, String username, String password) throws NamingException, JMSException {
+        this.destName = dest;
+        connectionFactory = (ConnectionFactory) context.lookup(connectionFactoryName);
+        connection = connectionFactory.createConnection(username, password);
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        if (isQueue) {
+            destination = session.createQueue(dest);
+        } else {
+            destination = session.createTopic(dest);
+        }
+        consumer = session.createConsumer(destination);
+        consumer.setMessageListener(this);
+        connection.start();
+    }
+    
+    /**
+     * The method used to handle incoming messages. Sends the formatted message to VANTIQ as a Source Notification.
+     */
+    @Override
+    public void onMessage(Message msg) {
+        Map<String, Object> msgMap = formatMessage(msg);
+        client.sendNotification(msgMap);
+    }
+    
+    /**
+     * A helper function to onMessage(), used to parse the incoming message and format it according to its message type
+     * @param message           The retrieved message from the queue
+     * @return                  A map containing the message, as well as the queue name and the JMS Message Type
+     * @throws JMSException
+     */
+    public Map<String, Object> formatMessage(Message message) {
+        Map<String, Object> msgMap = new LinkedHashMap<String, Object>();
+        
+        try {
+            if (message instanceof TextMessage) {
+                String msgText = ((TextMessage) message).getText();
+                msgMap.put("message", msgText);
+                msgMap.put("JMSFormat", TEXT);
+                msgMap.put("destination", this.destName);
+            } else if (message instanceof ObjectMessage) {
+                Object msgObject = ((ObjectMessage) message).getObject();
+                msgMap.put("message", msgObject);
+                msgMap.put("JMSFormat", OBJECT);
+                msgMap.put("destination", this.destName);
+            } else if (message instanceof MapMessage) {
+                
```

Comment:
Assuming these are things still to be done.

FYI:  Convention is generally to use `// FIXME: some description` or just `// FIXME`.  This indicates to the reviewer that you know stuff's missing, that it will be fixed before merge to master, etc.  As it is, things just fail silently which is the worst kind of failure.

For things that we should deal with but may do after merge/release, use `// TODO`.

IDE's recognize these and call them out (they show up as bright yellow for me, and are marked in the scroll bar).